File: SelectionSort.java
public class SelectionSort {
    public static void selectionSort(int[] A){
        for (int i = 0 ; i < A.length - 1 ; i++){
            int minIdx = i;
            for(int j = i+1 ; j < A.length ; j++){
                if (A[j] < A[minIdx]){
                    minIdx = j;
                }
            }

            int temp = A[minIdx];
            A[minIdx] = A[i];
            A[i] = temp;
        }
    }

    public static void printArray(int [] A){
        for(int i = 0 ; i < A.length ; i++){
            System.out.println(A[i] + " ");
        }
        System.out.println();
    }
    public static void main(String[] args) {

        int[] Arr = {4 , 6, 10,1 , 9 , 5};
        selectionSort(Arr);
        printArray(Arr);

    }
}

File: SelectionSort.cpp
#include <bits/stdc++.h>
using namespace std;

void selectionSort(int A[] , int size){
    
    for(int i = 0 ; i < size ; i++){
        int minEle = INT32_MAX;
        int minIndx = i;

        //Finding Minimum element
        for(int j = i ; j < size ; j++){
            
            if(A[j] < minEle){
                minEle = A[j];
                minIndx = j;
            }
        }

        //Swap
        int cpy = A[i];
        A[i] = minEle;
        A[minIndx] = cpy;
    }

    for(int i = 0 ; i < size ; i++){
        cout << A[i] << " ";
    }
}

int main(){

    int Arr[7] = {4 , 1 , 7 , 9 , 3 , 2 , 8};
    selectionSort(Arr , 7);
    cout << endl;

    return 0;
}